### PR TITLE
samples: debug: ppi_trace: use DEVICE_DT_GET

### PR DESCRIPTION
**Pull request by gmarull**
- Obtain RTC device reference at compile time
- Remove redundant Kconfig symbols (use DT status check instead)

### CODE REVIEW

File: samples/debug/ppi_trace/src/main.c [hunk 1]
```diff
@@ -79,7 +80,7 @@ static void alarm_callback(const struct device *dev, uint8_t chan_id,
 static void counter_setup(void)
 {
 	int err;
-	const struct device *dev = device_get_binding(RTC_LABEL);
+	const struct device *dev = DEVICE_DT_GET(RTC_NODE);
 
 	__ASSERT(dev, "Sample cannot run on this board.");
```
Comment (anangl) [inline samples/debug/ppi_trace/src/main.c:85]: This assertion should be replaced with a call to `device_is_ready`, as now `dev` will always be non-NULL.

Comment (gmarull) [reply samples/debug/ppi_trace/src/main.c]: oops good catch, fixed!

File: samples/debug/ppi_trace/src/main.c [hunk 2]
```diff
@@ -10,18 +10,19 @@
...
 #include <device.h>
+#include <devicetree.h>
 #include <logging/log.h>
 
 LOG_MODULE_REGISTER(app);
 
 #define ALARM_PERIOD_US 50000
 
-#if IS_ENABLED(CONFIG_USE_RTC2)
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(rtc2), okay)
```
Comment (anangl) [inline samples/debug/ppi_trace/src/main.c:20]: Since the `USE_RTC2` option is to stay, I'd rather drop this change, for consistency.

Comment (gmarull) [reply samples/debug/ppi_trace/src/main.c]: done

**Review by anangl — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**